examples/tls/envoy-http-https.yaml

```diff
@@ -0,0 +1,45 @@
+static_resources:
+  listeners:
+  - address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 10000
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          codec_type: auto
+          stat_prefix: ingress_http
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: app
+              domains:
+              - "*"
+              routes:
+              - match:
+                  prefix: "/"
+                route:
+                  cluster: service-https
+          http_filters:
+          - name: envoy.filters.http.router
+
+  clusters:
+  - name: service-https
+    connect_timeout: 0.25s
+    type: strict_dns
+    lb_policy: round_robin
+    load_assignment:
+      cluster_name: service-https
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: service-https
+                port_value: 443
+    transport_socket:
+      name: envoy.transport_sockets.tls
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
```

Comment:
This does no cert verification. I think we should demonstrate this as we would want to have verification of some kind?

Reply:
the double proxy does that

cert verification brings a load of other concerns - ie CA + certs, which is why i have avoided it here

i think perhaps i could make it clearer that this is the case and point to the double-proxy sandbox for example of mtls

Reply:
Yeah I just saw that. Can you add comments and/or link to the other example? I think we should be really clear that this is not a production ready setup.

Reply:
yep, lets try and land that one first then

Reply:
ive added some comments above which i will update once #13748 lands